src/main/proto/netflix/titus/titus_job_api.proto

```diff
@@ -258,6 +265,60 @@ message BasicContainer {
   // (Optional) A collection of system environment variables passed to the
   // container.
   map<string, string> env = 5;
+
+  // (Optional) An array of VolumeMounts. These VolumeMounts will be mounted in
+  // the container, and must reference one of the volumes declared for the Job.
+  // See the k8s docs
+  // https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.19/#volumemount-v1-core
+  // for more technical details.
+  repeated VolumeMount volumeMounts = 6;
+}
+
+// Volumes define some sort of storage for a Task (pod) that is later referenced
+// by individual containers via VolumeMount declarations.
+// https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.19/#volume-v1-core
+// Note that Titus only supports a subset of storage drivers.
+message Volume {
```

Comment:
This message defines only a name. How a volume resource reference (for example EBS) is provided?

Reply:
I only implemented `MainContainerVolume` type first.

I was planning on adding additional storage options here as they are implemented, but not before (to not confuse people to think that defining EBS here would actually work, which it would not)

Per https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.19/#volume-v1-core there are many many types, and I think we should only add them as they are actually supported.

Reply:
Might be worth documenting what the valid types are for name?

Reply:
For example, for different storage drivers we would have:

```
string name = 1;
// Configure this if you want to share folders between containers
SharedContainerVolumeSource sharedContainerVolumeSource = 2;
NFSVolumeSource nfsVolumeSource = 3;
EBSVolumeSource ebsVolumeSource = 4;
NDriveVolumeSource ?
```

Etc. This is one approach, and is how k8s does it (with classic volumes).

I don't anticipating us adding too many new driver configurations here.
Even though k8s does this approach, they have the catchall "CSIDriverVolumeSource" too, which is the freeform one, *in addition* to all the other volume types. Unlike k8s though, we  don't have to support arbitrary volume types.